src/main_app/jobs_workers/crop_main_files/process.py

```
@@ -266,60 +272,119 @@ def process_crops(
+        if not file_info["steps"]["crop"]["result"]:
+            result["summary"]["failed"] += 1
+            file_info["status"] = "failed"
+            file_info["error"] = file_info["steps"]["crop"]["msg"]
+        else:
+            result["summary"]["cropped"] += 1
+            file_info["cropped_path"] = cropped_output_path
+
         if not upload_files:
             file_info["status"] = "skipped"
-            file_info["reason"] = "upload disabled"
+            file_info["steps"]["upload_cropped"] = {"result": None, "msg": "Skipped – upload disabled"}
```

Ruff (0.15.2)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)